### PR TITLE
Remove internal pause implementation

### LINKED ISSUE #2993: Static Linking Not Supported in RHEL 8
When building cri-o from source, it enforces static linking for example in the pause/Makefile:

```
src = $(wildcard *.c)
obj = $(src:.c=.o)
override LIBS +=
CFLAGS ?= -std=c99 -Os -Wall -Wextra
override CFLAGS += -static
../bin/pause: $(obj) | ../bin
        $(CC) -o $@ $^ $(CFLAGS) $(LIBS)
../bin:
        mkdir -p $@
.PHONY: clean
clean:
        rm -f $(obj) ../bin/pause
```

However, static linking is not supported in RHEL 8: https://access.redhat.com/articles/rhel8-abi-compatibility

```
Static Linking with the C/C++ Runtime
Static linking with the C/C++ runtime is not supported. This includes linking with any files that are part of the glibc-static or libstdc++-static packages. You may choose to link statically, but the resulting application binaries may fail to operate if any package in the installation is changed.
```

This means that glibc-static is not available for install in RHEL 8. We can build on RHEL 8 by commenting out the lines that force a static build:

```
- override CFLAGS += -static
+ #override CFLAGS += -static
```

**Steps to reproduce the issue:**
1. Run make to build cri-o on RHEL 8

**Describe the results you received:**
```
[ec2-user@a cri-o]$ make
mkdir -p "/home/ec2-user/cri-o/_output/src/github.com/cri-o"
ln -s "/home/ec2-user/cri-o" "/home/ec2-user/cri-o/_output/src/github.com/cri-o/cri-o"
touch "/home/ec2-user/cri-o/_output/.gopathok"
GO111MODULE=on go build -mod=vendor -ldflags '-s -w -X main.gitCommit="0c5540dabeba426e514e870754b395e8569a805b-dirty" -X main.buildInfo=1574741461' -tags "containers_image_ostree_stub  exclude_graphdriver_btrfs btrfs_noversion exclude_graphdriver_devicemapper libdm_no_deferred_remove containers_image_openpgp  " -o bin/crio github.com/cri-o/cri-o/cmd/crio
make -C pause
make[1]: Entering directory '/home/ec2-user/cri-o/pause'
cc -std=c99 -Os -Wall -Wextra -static   -c -o pause.o pause.c
cc -o ../bin/pause pause.o -std=c99 -Os -Wall -Wextra -static 
/usr/bin/ld: cannot find -lc
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:9: ../bin/pause] Error 1
make[1]: Leaving directory '/home/ec2-user/cri-o/pause'
make: *** [Makefile:117: bin/pause] Error 2
```

**Describe the results you expected:**
I expect cri-o to build successfully without having to manually edit the Makefiles. Perhaps option we can pass to the make command that enables static vs dynamic linking?

**Additional information you deem important (e.g. issue happens only occasionally):**
Static linking does not seem to be supported in RHEL 8

**Output of `crio --version`:**

```
N/A - trying to build from source
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
AWS EC2 RHEL 8.0

### PR DESCRIPTION
**Pull request by saschagrunert**
The pause binary has only be used for testing purposes and has no
need in production use cases. We now remove it and adapt the testing
around to avoid improper usage.

Closes #2993, #2444

### CODE REVIEW

**Comment by haircommander** [comment]
there's a stage of helpers.bash tests where we prepull images. We should also prepull pause

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Comment by saschagrunert** [comment]
> there's a stage of helpers.bash tests where we prepull images. We should also prepull pause

Yeah I will change that as well.

File: test/helpers.bash [hunk 1]
```diff
@@ -97,6 +93,16 @@ if ! [ -d "$ARTIFACTS_PATH"/redis-image ]; then
 	fi
 fi
 
+# Make sure we have a copy of the redis:alpine image.
```
Comment (haircommander) [inline test/helpers.bash]: nit: s/redis:alpine/pause

Comment (saschagrunert) [reply test/helpers.bash]: Fixed :+1:

**Comment by haircommander** [comment]
LGTM

**Comment by haircommander** [comment]
/retest
@umohnani8 @mrunalp PTAL

**Pull request merged**